`indentation_width` should exclude indentation within multiline strings

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`indentation_width` probably shouldn't flag content in multiline strings, as it's likely not Swift code and might follow different indentation norms.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint test.swift --config indent.yml
Linting Swift files at paths test.swift
Linting 'test.swift' (1/1)
test.swift:3:1: warning: Indentation Width Violation: Code should be indented using one tab or 4 spaces. (indentation_width)
Done linting! Found 1 violation, 0 serious in 1 file.

$ cat test.swift
let jsonFixture = """
{
  "test": true
}
"""
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.49.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? building from source
* Paste your configuration file:

```yml
opt_in_rules:
  indentation_width
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -86,6 +86,7 @@
 * Improve `identifier_name` documentation.  
   [Martin Redington](https://github.com/mildm8nnered)
   [#4767](https://github.com/realm/SwiftLint/issues/4767)
+
 * Adds `include_multiline_strings` option to `indentation_width` rule.  
   [Martin Redington](https://github.com/mildm8nnered)
   [#4248](https://github.com/realm/SwiftLint/issues/4248)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -24,7 +24,7 @@
 
 * Make forceExclude work with directly specified files.  
   [jimmya](https://github.com/jimmya)
-  [#issue_number](https://github.com/realm/SwiftLint/issues/4609)
+  [#4609](https://github.com/realm/SwiftLint/issues/4609)
 
 * Separate analyzer rules as an independent section in the rule directory of
   the reference.  
@@ -86,6 +86,9 @@
 * Improve `identifier_name` documentation.  
   [Martin Redington](https://github.com/mildm8nnered)
   [#4767](https://github.com/realm/SwiftLint/issues/4767)
+* Adds `include_multiline_strings` option to `indentation_width` rule.  
+  [Martin Redington](https://github.com/mildm8nnered)
+  [#4248](https://github.com/realm/SwiftLint/issues/4248)
 
 #### Bug Fixes
 

--- a/Source/SwiftLintFramework/Rules/RuleConfigurations/IndentationWidthConfiguration.swift
+++ b/Source/SwiftLintFramework/Rules/RuleConfigurations/IndentationWidthConfiguration.swift
@@ -4,23 +4,27 @@ struct IndentationWidthConfiguration: RuleConfiguration, Equatable {
             + "indentation_width: \(indentationWidth), "
             + "include_comments: \(includeComments), "
             + "include_compiler_directives: \(includeCompilerDirectives)"
+            + "include_multiline_strings: \(includeMultilineStrings)"
     }
 
     private(set) var severityConfiguration: SeverityConfiguration
     private(set) var indentationWidth: Int
     private(set) var includeComments: Bool
     private(set) var includeCompilerDirectives: Bool
+    private(set) var includeMultilineStrings: Bool
 
     init(
         severity: ViolationSeverity,
         indentationWidth: Int,
         includeComments: Bool,
-        includeCompilerDirectives: Bool
+        includeCompilerDirectives: Bool,
+        includeMultilineStrings: Bool
     ) {
         self.severityConfiguration = SeverityConfiguration(severity)
         self.indentationWidth = indentationWidth
         self.includeComments = includeComments
         self.includeCompilerDirectives = includeCompilerDirectives
+        self.includeMultilineStrings = includeMultilineStrings
     }
 
     mutating func apply(configuration: Any) throws {
@@ -42,6 +46,10 @@ struct IndentationWidthConfiguration: RuleConfiguration, Equatable {
 
         if let includeCompilerDirectives = configurationDict["include_compiler_directives"] as? Bool {
             self.includeCompilerDirectives = includeCompilerDirectives
+        }
+        
+        if let includeMultilineStrings = configurationDict["include_multiline_strings"] as? Bool {
+            self.includeMultilineStrings = includeMultilineStrings
         }
     }
 }

--- a/Source/SwiftLintFramework/Rules/RuleConfigurations/IndentationWidthConfiguration.swift
+++ b/Source/SwiftLintFramework/Rules/RuleConfigurations/IndentationWidthConfiguration.swift
@@ -47,7 +47,7 @@ struct IndentationWidthConfiguration: RuleConfiguration, Equatable {
         if let includeCompilerDirectives = configurationDict["include_compiler_directives"] as? Bool {
             self.includeCompilerDirectives = includeCompilerDirectives
         }
-        
+
         if let includeMultilineStrings = configurationDict["include_multiline_strings"] as? Bool {
             self.includeMultilineStrings = includeMultilineStrings
         }

--- a/Source/SwiftLintFramework/Rules/Style/IndentationWidthRule.swift
+++ b/Source/SwiftLintFramework/Rules/Style/IndentationWidthRule.swift
@@ -20,7 +20,8 @@ struct IndentationWidthRule: ConfigurationProviderRule, OptInRule {
         severity: .warning,
         indentationWidth: 4,
         includeComments: true,
-        includeCompilerDirectives: true
+        includeCompilerDirectives: true,
+        includeMultilineStrings: true
     )
     static let description = RuleDescription(
         identifier: "indentation_width",
@@ -34,6 +35,7 @@ struct IndentationWidthRule: ConfigurationProviderRule, OptInRule {
             Example("firstLine\n\tsecondLine\n\t\tthirdLine\n\n\t\tfourthLine"),
             Example("firstLine\n\tsecondLine\n\t\tthirdLine\n\t//test\n\t\tfourthLine"),
             Example("firstLine\n    secondLine\n        thirdLine\nfourthLine")
+//            Example("let x = \"\"\"\nstring1\n    string2\n  string3\n\"\"\"\n")
         ],
         triggeringExamples: [
             Example("↓    firstLine", testMultiByteOffsets: false, testDisableCommand: false),
@@ -59,6 +61,8 @@ struct IndentationWidthRule: ConfigurationProviderRule, OptInRule {
             if line.content.count == indentationCharacterCount { continue }
 
             if ignoreComment(line: line, in: file) { continue }
+
+            if ignoreMultilineStrings(line: line, in: file) { continue }
 
             // Get space and tab count in prefix
             let prefix = String(line.content.prefix(indentationCharacterCount))
@@ -162,6 +166,16 @@ struct IndentationWidthRule: ConfigurationProviderRule, OptInRule {
         }
         let syntaxKindsInLine = Set(file.syntaxMap.tokens(inByteRange: line.byteRange).kinds)
         if syntaxKindsInLine.isNotEmpty, SyntaxKind.commentKinds.isSuperset(of: syntaxKindsInLine) {
+            return true
+        }
+        return false
+    }
+    
+    private func ignoreMultilineStrings(line: Line, in file: SwiftLintFile) -> Bool {
+        if configuration.includeMultilineStrings {
+            return false
+        }
+        if file.syntaxMap.tokens(inByteRange: line.byteRange).kinds == [.string] {
             return true
         }
         return false

--- a/Source/SwiftLintFramework/Rules/Style/IndentationWidthRule.swift
+++ b/Source/SwiftLintFramework/Rules/Style/IndentationWidthRule.swift
@@ -60,9 +60,7 @@ struct IndentationWidthRule: ConfigurationProviderRule, OptInRule {
             let indentationCharacterCount = line.content.countOfLeadingCharacters(in: CharacterSet(charactersIn: " \t"))
             if line.content.count == indentationCharacterCount { continue }
 
-            if ignoreComment(line: line, in: file) { continue }
-
-            if ignoreMultilineStrings(line: line, in: file) { continue }
+            if ignoreComment(line: line, in: file) || ignoreMultilineStrings(line: line, in: file) { continue }
 
             // Get space and tab count in prefix
             let prefix = String(line.content.prefix(indentationCharacterCount))
@@ -170,7 +168,7 @@ struct IndentationWidthRule: ConfigurationProviderRule, OptInRule {
         }
         return false
     }
-    
+
     private func ignoreMultilineStrings(line: Line, in file: SwiftLintFile) -> Bool {
         if configuration.includeMultilineStrings {
             return false

--- a/Source/SwiftLintFramework/Rules/Style/IndentationWidthRule.swift
+++ b/Source/SwiftLintFramework/Rules/Style/IndentationWidthRule.swift
@@ -35,7 +35,6 @@ struct IndentationWidthRule: ConfigurationProviderRule, OptInRule {
             Example("firstLine\n\tsecondLine\n\t\tthirdLine\n\n\t\tfourthLine"),
             Example("firstLine\n\tsecondLine\n\t\tthirdLine\n\t//test\n\t\tfourthLine"),
             Example("firstLine\n    secondLine\n        thirdLine\nfourthLine")
-//            Example("let x = \"\"\"\nstring1\n    string2\n  string3\n\"\"\"\n")
         ],
         triggeringExamples: [
             Example("↓    firstLine", testMultiByteOffsets: false, testDisableCommand: false),

--- a/Tests/SwiftLintFrameworkTests/IndentationWidthRuleTests.swift
+++ b/Tests/SwiftLintFrameworkTests/IndentationWidthRuleTests.swift
@@ -196,12 +196,27 @@ class IndentationWidthRuleTests: XCTestCase {
             """, includeCompilerDirectives: true)
     }
 
+    func testIgnoredMultilineStrings() {
+        assertNoViolation(
+            in: "let x = \"\"\"\nstring1\n    string2\n  string3\n\"\"\"\n",
+            includeMultilineStrings: false
+        )
+        assert1Violation(
+            in: "let x = \"\"\"\nstring1\n    string2\n  string3\n\"\"\"\n"
+        )
+        assertViolations(
+            in: "let x = \"\"\"\nstring1\n    string2\n  string3\n string4\n\"\"\"\n",
+            equals: 2
+        )
+    }
+
     // MARK: Helpers
     private func countViolations(
         in example: Example,
         indentationWidth: Int? = nil,
         includeComments: Bool? = nil,
         includeCompilerDirectives: Bool? = nil,
+        includeMultilineStrings: Bool? = nil,
         file: StaticString = #file,
         line: UInt = #line
     ) -> Int {
@@ -214,6 +229,9 @@ class IndentationWidthRuleTests: XCTestCase {
         }
         if let includeCompilerDirectives {
             configDict["include_compiler_directives"] = includeCompilerDirectives
+        }
+        if let includeMultilineStrings {
+            configDict["include_multiline_strings"] = includeMultilineStrings
         }
 
         guard let config = makeConfig(configDict, IndentationWidthRule.description.identifier) else {
@@ -230,6 +248,7 @@ class IndentationWidthRuleTests: XCTestCase {
         indentationWidth: Int? = nil,
         includeComments: Bool? = nil,
         includeCompilerDirectives: Bool? = nil,
+        includeMultilineStrings: Bool? = nil,
         file: StaticString = #file,
         line: UInt = #line
     ) {
@@ -239,6 +258,7 @@ class IndentationWidthRuleTests: XCTestCase {
                 indentationWidth: indentationWidth,
                 includeComments: includeComments,
                 includeCompilerDirectives: includeCompilerDirectives,
+                includeMultilineStrings: includeMultilineStrings,
                 file: file,
                 line: line
             ),
@@ -253,6 +273,7 @@ class IndentationWidthRuleTests: XCTestCase {
         indentationWidth: Int? = nil,
         includeComments: Bool? = nil,
         includeCompilerDirectives: Bool? = nil,
+        includeMultilineStrings: Bool? = nil,
         file: StaticString = #file,
         line: UInt = #line
     ) {
@@ -262,6 +283,7 @@ class IndentationWidthRuleTests: XCTestCase {
             indentationWidth: indentationWidth,
             includeComments: includeComments,
             includeCompilerDirectives: includeCompilerDirectives,
+            includeMultilineStrings: includeMultilineStrings,
             file: file,
             line: line
         )
@@ -272,6 +294,7 @@ class IndentationWidthRuleTests: XCTestCase {
         indentationWidth: Int? = nil,
         includeComments: Bool? = nil,
         includeCompilerDirectives: Bool? = nil,
+        includeMultilineStrings: Bool? = nil,
         file: StaticString = #file,
         line: UInt = #line
     ) {
@@ -281,6 +304,7 @@ class IndentationWidthRuleTests: XCTestCase {
             indentationWidth: indentationWidth,
             includeComments: includeComments,
             includeCompilerDirectives: includeCompilerDirectives,
+            includeMultilineStrings: includeMultilineStrings,
             file: file,
             line: line
         )

